src/index.js

```diff
@@ -1,228 +1,4 @@
-import EventEmitter from 'events';
-import log from 'apify-shared/log';
-import { main, getEnv, call, callTask, getApifyProxyUrl, metamorph, addWebhook } from './actor';
-import AutoscaledPool from './autoscaling/autoscaled_pool';
-import BasicCrawler from './crawlers/basic_crawler';
-import CheerioCrawler from './crawlers/cheerio_crawler';
-import { pushData, openDataset } from './dataset';
-import events, { initializeEvents, stopEvents } from './events';
-import { getValue, setValue, getInput, openKeyValueStore } from './key_value_store';
-import { launchPuppeteer } from './puppeteer';
-import PuppeteerCrawler from './crawlers/puppeteer_crawler';
-import PuppeteerPool from './puppeteer_pool';
-import Request from './request';
-import { RequestList, openRequestList } from './request_list';
-import { openRequestQueue } from './request_queue';
-import { apifyClient, getMemoryInfo, isAtHome, publicUtils } from './utils';
-import { puppeteerUtils } from './puppeteer_utils';
-import { socialUtils } from './utils_social';
-import { enqueueLinks } from './enqueue_links/enqueue_links';
-import PseudoUrl from './pseudo_url';
-import LiveViewServer from './live_view/live_view_server';
-import { requestAsBrowser } from './utils_request';
-import { openSessionPool } from './session_pool/session_pool';
-import { Session } from './session_pool/session';
+import * as Apify from './index.esm';
```

Comment:
strictly speaking, `import Apify from 'apify'` means `require('apify').default` (as per ESM itself), since it doesn't export a default, so comes out as undefined. you need to either duplicate the current exports on a `export default exports` (for babel to reflect the build correctly and the module interop to work) or help node know which format we are using, specifying a "module" and pointing to a esm that does the same exports, so that you don't need to change the main entrypoint (that will be build/index.js)